src/main/java/io/github/thebusybiscuit/slimefun4/core/services/profiler/SlimefunProfiler.java

```diff
@@ -0,0 +1,251 @@
+                }
+            }
+
+            if (running.get()) {
+                // Looks like the next profiling has already started, abort!
+                return;
+            }
+
+            totalElapsedTime = timings.values().stream().mapToLong(Long::longValue).sum();
+
+            if (!requests.isEmpty()) {
+                PerformanceSummary summary = new PerformanceSummary(this, totalElapsedTime, timings.size());
+                Iterator<CommandSender> iterator = requests.iterator();
+
+                while (iterator.hasNext()) {
+                    summary.send(iterator.next());
+                    iterator.remove();
+                }
+            }
+        });
+
+    }
+
+    /**
+     * This method requests a summary for the given {@link CommandSender}.
+     * The summary will be sent upon the next available moment in time.
+     * 
+     * @param sender
+     *            The {@link CommandSender} who shall receive this summary.
+     */
+    public void requestSummary(CommandSender sender) {
+        requests.add(sender);
+    }
+
+    protected Map<String, Long> getByItem() {
+        Map<String, Long> map = new HashMap<>();
+
+        for (Map.Entry<ProfiledBlock, Long> entry : timings.entrySet()) {
+            map.merge(entry.getKey().getId(), entry.getValue(), Long::sum);
+        }
+
+        return map;
+    }
+
+    protected Map<String, Long> getByChunk() {
+        Map<String, Long> map = new HashMap<>();
+
+        for (Map.Entry<ProfiledBlock, Long> entry : timings.entrySet()) {
+            String world = entry.getKey().getPosition().getWorld().getName();
+            int x = entry.getKey().getPosition().getChunkX();
+            int z = entry.getKey().getPosition().getChunkZ();
+
+            map.merge(world + " (" + x + ',' + z + ')', entry.getValue(), Long::sum);
+        }
+
+        return map;
+    }
+
+    protected int getBlocksInChunk(String chunk) {
+        int blocks = 0;
+
+        for (ProfiledBlock block : timings.keySet()) {
+            String world = block.getPosition().getWorld().getName();
+            int x = block.getPosition().getChunkX();
+            int z = block.getPosition().getChunkZ();
+
+            if (chunk.equals(world + " (" + x + ',' + z + ')')) {
+                blocks++;
+            }
+        }
+
+        return blocks;
+    }
+
+    protected int getBlocksOfId(String id) {
+        int blocks = 0;
+
+        for (ProfiledBlock block : timings.keySet()) {
+            if (block.getId().equals(id)) {
+                blocks++;
+            }
+        }
+
+        return blocks;
+    }
+
+    /**
+     * This method returns the current {@link PerformanceRating}.
+     * 
+     * @return The current performance grade
+     */
+    public PerformanceRating getPerformance() {
+        float percentage = Math.round(((((totalElapsedTime / 1000000.0) * 100.0F) / PerformanceSummary.MAX_TICK_DURATION) * 100.0F) / 100.0F);
```

Comment:
Again this in the one line looks really cumbersome. Also the fact it's used in at least 2 places means this should probably be a util method. 

Would make this clearer and avoid human error which is definitely possible for something like this.